cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -162,20 +163,22 @@ private String getSpecJson(Application app, final String bundleVersion, final St
       configurer = artifactId == null ?
         new DefaultAppConfigurer(app, configString) :
         new DefaultAppConfigurer(artifactId, app, configString, artifactRepository, pluginInstantiator);
-    }
 
-    Config appConfig;
-    TypeToken typeToken = TypeToken.of(app.getClass());
-    TypeToken<?> configToken = typeToken.resolveType(Application.class.getTypeParameters()[0]);
-    if (Strings.isNullOrEmpty(configString)) {
-      appConfig = (Config) configToken.getRawType().newInstance();
-    } else {
-      //TODO: CDAP-2869 Handle JsonSyntax Exception better and throw a more meaningful error. This will be done when we
-      //use PluginInstantiator methods instead of GSON deserialization
-      appConfig = GSON.fromJson(configString, configToken.getType());
-    }
+      Config appConfig;
```

Comment:
Oh wondering how it was working before. Because we were creating the PluginInstantiator in ``try`` but we are constructing only the Configurer. But the plugin registration is being done only when we call the application's configure method. But PluginInstantiator resource would have been closed by then. So moving that logic inside the try block.